monai/apps/auto3dseg/algorithms/templates/segresnet2d/scripts/infer.py

```diff
@@ -0,0 +1,252 @@
+# limitations under the License.
+
+import json
+import logging
+import os
+import sys
+from typing import Optional, Sequence, Union
+
+import torch
+
+import monai
+from monai import transforms
+from monai.bundle import ConfigParser
+from monai.bundle.scripts import _pop_args, _update_args
+from monai.data import ThreadDataLoader, decollate_batch, list_data_collate
+from monai.inferers import sliding_window_inference
+
+
+class InferClass:
+    def __init__(self, config_file: Optional[Union[str, Sequence[str]]] = None, **override):
+        logging.basicConfig(stream=sys.stdout, level=logging.INFO)
+
+        _args = _update_args(config_file=config_file, **override)
+        config_file_ = _pop_args(_args, "config_file")[0]
+
+        parser = ConfigParser()
+        parser.read_config(config_file_)
+        parser.update(pairs=_args)
+
+        data_file_base_dir = parser.get_parsed_content("data_file_base_dir")
+        data_list_file_path = parser.get_parsed_content("data_list_file_path")
+        self.fast = parser.get_parsed_content("infer")["fast"]
+        self.num_adjacent_slices = parser.get_parsed_content("num_adjacent_slices")
+        self.num_sw_batch_size = parser.get_parsed_content("num_sw_batch_size")
+        self.output_classes = parser.get_parsed_content("output_classes")
+        self.overlap_ratio = parser.get_parsed_content("overlap_ratio")
+        self.patch_size_valid = parser.get_parsed_content("patch_size_valid")
+        softmax = parser.get_parsed_content("softmax")
+
+        ckpt_name = parser.get_parsed_content("infer")["ckpt_name"]
+        data_list_key = parser.get_parsed_content("infer")["data_list_key"]
+        output_path = parser.get_parsed_content("infer")["ouptut_path"]
+
+        if not os.path.exists(output_path):
+            os.makedirs(output_path, exist_ok=True)
+
+        self.infer_transforms = parser.get_parsed_content("transforms_infer")
+
+        with open(data_list_file_path) as f:
+            json_data = json.load(f)
+
+        list_data = []
+        for item in json_data[data_list_key]:
+            list_data.append(item)
+
+        files = []
+        for _i in range(len(list_data)):
+            str_img = os.path.join(data_file_base_dir, list_data[_i]["image"])
+
+            if not os.path.exists(str_img):
+                continue
+
+            files.append({"image": str_img})
+
+        self.infer_files = files
+
+        self.infer_loader = None
+        if self.fast:
+            infer_ds = monai.data.Dataset(data=self.infer_files, transform=self.infer_transforms)
+            self.infer_loader = ThreadDataLoader(infer_ds, num_workers=8, batch_size=1, shuffle=False)
+
+        self.device = torch.device("cuda:0")
+        torch.cuda.set_device(self.device)
+
+        self.model = parser.get_parsed_content("network")
+        self.model = self.model.to(self.device)
+        self.model = torch.nn.SyncBatchNorm.convert_sync_batchnorm(self.model)
```

Comment:
Same as my comment in infer.py, SyncBatchNorm caused some issue when I trained with one GPU